modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/Transactions/ClientTransaction.cs

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Core.Impl.Client.Transactions
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Ignite Thin Client transaction facade.
+    /// </summary>
+    internal class ClientTransaction: IClientTransactionInternal
+    {
+        /** Unique  transaction ID.*/
+        private readonly int _id;
+
+        /** Ignite. */
+        private readonly IgniteClient _ignite;
+
+        /* Transactions. */
+        private readonly ClientTransactions _transactions;
+
+        /** Transaction is closed. */
+        private volatile bool _closed; 
```

Comment:
Do we need `volatile`? Transactions are tied to a specific thread, right?

Reply:
In theory transaction returned by `TxStart` can be closed from any thread.

Reply:
Usually, types and APIs are not thread safe, unless stated otherwise. E.g. we don't expect `List<T>` to be thread safe and have volatile fields. We explicitly state that Ignite transactions are per-thread, so we don't need to guarantee thread safety.